src/NServiceBus.Core/Receiving/ReceiveComponentFactory.cs

```diff
@@ -0,0 +1,84 @@
+namespace NServiceBus
+{
+    using System;
+    using System.Threading.Tasks;
+    using ObjectBuilder;
+    using Routing;
+    using Settings;
+    using Transport;
+
+    static class ReceiveComponentFactory
+    {
+        public static ReceiveConfiguration Configure(string endpointName, bool isSendOnlyEndpoint, TransportInfrastructure transportInfrastructure, ReadOnlySettings settings)
+        {
+            if (isSendOnlyEndpoint)
+            {
+                return new ReceiveConfiguration(new LogicalAddress(), null, null, null, TransportTransactionMode.None, PushRuntimeSettings.Default, false, false);
+            }
+
+            var discriminator = settings.GetOrDefault<string>("EndpointInstanceDiscriminator");
+            var queueNameBase = settings.GetOrDefault<string>("BaseInputQueueName") ?? endpointName;
+            var purgeOnStartup = settings.GetOrDefault<bool>("Transport.PurgeOnStartup");
+
+            //note: This is an old hack, we are passing the endpoint name to bind but we only care about the properties
+            var mainInstanceProperties = transportInfrastructure.BindToLocalEndpoint(new EndpointInstance(endpointName)).Properties;
+
+            var logicalAddress = LogicalAddress.CreateLocalAddress(queueNameBase, mainInstanceProperties);
+
+            var localAddress = transportInfrastructure.ToTransportAddress(logicalAddress);
+
+            string instanceSpecificQueue = null;
+            if (discriminator != null)
+            {
+                instanceSpecificQueue = transportInfrastructure.ToTransportAddress(logicalAddress.CreateIndividualizedAddress(discriminator));
+            }
+
+            var transactionMode = GetRequiredTransactionMode(settings);
+
+            var pushRuntimeSettings = GetDequeueLimitations(settings);
+
+            var config = new ReceiveConfiguration(logicalAddress, queueNameBase, localAddress, instanceSpecificQueue, transactionMode, pushRuntimeSettings, purgeOnStartup, true);
+
+            settings.AddStartupDiagnosticsSection("Receiving", config);
```

Comment:
this means there is no receiving section at all when running in send-only. I guess that's ok, but since the component itself currently always "runs" bus has an "enabled" state, isn't the "readonly" mode also something we'd want to state? (although the absence of a receiving section probably says everything?)

Reply:
I've added it back so in send only mode we will write:

```

"Receiving":{
   "LogicalAddress":{
      "Qualifier":null,
      "EndpointInstance":null
   },
   "LocalAddress":null,
   "InstanceSpecificQueue":null,
   "TransactionMode":0,
   "PushRuntimeSettings":{
      "MaxConcurrency":8
   },
   "QueueNameBase":null,
   "IsEnabled":false,
   "SatelliteDefinitions":{
      "Definitions":[

      ]
   }
```